### PR TITLE
Introduce Base and JSON log formatters

### PR DESCRIPTION
**Pull request by davydovanton**
Reference to https://github.com/hanami/hanami/issues/483

### CODE REVIEW

File: lib/hanami/logger.rb [hunk 1]
```diff
@@ -91,8 +92,33 @@ class Formatter < ::Logger::Formatter
       #
       # @see http://www.ruby-doc.org/stdlib/libdoc/logger/rdoc/Logger/Formatter.html#method-i-call
       def call(severity, time, progname, msg)
-        progname = "[#{@application_name}] #{progname}"
-        super(severity, time.utc, progname, msg)
+        hash = {
+          application_name: @application_name,
```
Comment (jodosha) [inline lib/hanami/logger.rb]: @davydovanton What about a shorter `app` instead of `application_name`?

Comment (davydovanton) [inline lib/hanami/logger.rb]: good idea :+1:

File: lib/hanami/logger.rb [hunk 2]
```diff
@@ -91,8 +92,33 @@ class Formatter < ::Logger::Formatter
...
+          severity: severity,
+          time: time.utc,
+          progname: progname
```
Comment (jodosha) [inline lib/hanami/logger.rb]: @davydovanton Is `progname` still used and/or useful? It's always null in the tests.

File: lib/hanami/logger.rb [hunk 3]
```diff
@@ -91,8 +92,33 @@ class Formatter < ::Logger::Formatter
...
+      # @api private
+      def _message_hash(message)
+        case message
+        when Hash
+          message
+        when Exception
+          {
+            message: message.message,
+            backtrace: message.backtrace || [],
+            error_class: message.class
```
Comment (jodosha) [inline lib/hanami/logger.rb]: @davydovanton what about `error`, instead of `error_class`?

File: lib/hanami/logger.rb (hunk 2 above)
Comment (davydovanton) [inline lib/hanami/logger.rb]: So, in all my tests and projects `progname` was null. I'll remove this

**Comment by davydovanton** [comment]
@jodosha done. I renamed keys and deleted `progname` unnecessary value :tada:

**Comment by davydovanton** [comment]
@jodosha updated documentation examples and added part about formatters 👍

**Comment by jodosha** [comment]
@davydovanton Thank you very much. Just merged! 👍

**Comment by davydovanton** [comment]
amazing 🎉 
Thanks for helping 👍

**Comment by joneslee85** [comment]
Great work. Thanks @davydovaton

**Pull request merged**